pkg/sip/server.go

```diff
@@ -132,8 +139,10 @@ type Server struct {
 	sipListeners []io.Closer
 	sipUnhandled RequestHandler
 
-	imu               sync.Mutex
-	inProgressInvites []*inProgressInvite
+	imu                  sync.Mutex
+	inProgressInvites    map[dialogKey]*inProgressInvite
+	inviteTimeoutQueue   utils.TimeoutQueue[*dialogKey]
```

Comment:
There's a convention in Go called a [Mutex hat](https://dmitri.shuralyov.com/idiomatic-go/entries/2), which implies `inviteTimeoutQueue` is protected by `imu`.

But, the queue implementation already has a mutex internally. So might be worth moving it out of the `imu` group.